orix/io/plugins/ctf.py

```diff
@@ -0,0 +1,357 @@
+    n_cols_expected = [len(column_names[vendor][k]) for k in range(n_variants)]
+    if vendor == "orix" and "Column names" in footprint_line:
+        # Append names of extra properties found, if any, in the orix
+        # .ang file header
+        vendor_column_names = column_names[vendor][0]
+        n_cols = n_cols_expected[0]
+        extra_props = footprint_line.split(":")[1].split(",")[n_cols:]
+        vendor_column_names += [i.lstrip(" ").replace(" ", "_") for i in extra_props]
+    elif n_cols_file not in n_cols_expected:
+        warnings.warn(
+            f"Number of columns, {n_cols_file}, in the file is not equal to "
+            f"the expected number of columns, {n_cols_expected}, for the \n"
+            f"assumed vendor '{vendor}'. Will therefore assume the following "
+            "columns: phase_id, x, y, bands, error, euler1, euler2, euler3"
+            "MAD, BC, BS, etc."
+        )
+        vendor = "unknown"
+        vendor_column_names = column_names[vendor][0]
+        n_cols = len(vendor_column_names)
+        if n_cols_file > n_cols:
+            # Add any extra columns as properties
+            for i in range(n_cols_file - n_cols):
+                vendor_column_names.append("unknown" + str(i + 3))
+    else:
+        idx = np.where(np.equal(n_cols_file, n_cols_expected))[0][0]
+        vendor_column_names = column_names[vendor][idx]
+
+    return vendor, vendor_column_names
+
+
+def _get_phases_from_header(
+    header: List[str],
+) -> Tuple[List[int], List[str], List[str], List[List[float]]]:
+    """Return phase names and symmetries detected in an .ctf file
+    header.
+
+    Parameters
+    ----------
+    header
+        List with header lines as individual elements.
+
+    Returns
+    -------
+    ids
+        Phase IDs.
+    phase_names
+        List of names of detected phases.
+    phase_point_groups
+        List of point groups of detected phase.
+    lattice_constants
+        List of list of lattice parameters of detected phases.
+
+    Notes
+    -----
+    Regular expressions are used to collect phase name, formula and
+    point group. This function have been tested with files from the
+    following vendor's formats: Oxford AZtec HKL, and EMsoft v4/v5.
+    """
+
+    phases = {
+        "name": [],
+        "space_group": [],
+        "lattice_constants": [],
+        "id": [],
+    }
+    phase_num_row = 0
+    for line in header:
+        if "Phases" in line:
+            phases_num_line = line
+            break
+        phase_num_row += 1
+    phase_num = int(phases_num_line.split("\t")[1])
+
+    for num in range(phase_num):
+        phase_data = header[phase_num_row + num + 1].split("\t")
+        phases["name"].append(phase_data[2])
+        phases["space_group"].append(int(phase_data[4]))
+        phases["lattice_constants"].append(
+            [float(i) for i in phase_data[0].split(";") + phase_data[1].split(";")]
+        )
+        phases["id"].append(num + 1)
```

Comment:
This can be simplified a bit to:

```suggestion
    for i, line in enumerate(header):
        if line.startswith("Phases"):
            break

    n_phases = int(line.split("\t")[1])

    for j in range(n_phases):
        phase_data = header[i + 1 + j].split("\t")
        phases["name"].append(phase_data[2])
        phases["space_group"].append(int(phase_data[4]))
        phases["lattice_constants"].append(
            [float(i) for i in phase_data[0].split(";") + phase_data[1].split(";")]
        )
        phases["id"].append(j + 1)
```